### ISSUE TITLE
how to generate a tumor model?

### ISSUE THREAD

**Issue by zhangyafeng1** [BostonGene/Kassandra#13]
1. **model_column = 'Tumor_model_annot'**
2. samples = data_annot.loc[data_annot['Tumor_model_annot'] == 'cancer_cells'].index
3. cancer_expr = data_expr[samples]  
4. cancer_annot = data_annot.loc[samples]
5. cancer_annot['Tumor_type'] = cancer_annot['Dataset']
6. cancer_annot = cancer_annot[['Tumor_type', 'Dataset']]
7. 
8. **samples = data_annot.loc[~data_annot[model_column].isna() &   (data_annot['Tumor_model_annot'] != 'cancer_cells')].index** 
9. 
10. cells_expr = data_expr[samples]
11. cells_annot = data_annot.loc[samples] 
12. cells_annot = cells_annot[[model_column, 'Dataset']]
13. cells_annot.columns = ['Cell_type', 'Dataset'] 
14. cells_annot = pd.concat([lab_annot, cells_annot])
15. cells_annot.loc[cells_annot['Dataset'].isna(), 'Dataset'] = cells_annot.loc[cells_annot['Dataset'].isna()].index
16. cells_expr = pd.concat([lab_expr, cells_expr], axis=1)
17. 
18. # to make sure that there is no repeated samples
19. samples = sorted(list(set(cells_annot.index).intersection(set(cells_expr.columns))))
20. cells_expr = cells_expr[samples]
21. cells_annot = cells_annot.loc[samples]
22. 
23. print(cells_expr.shape, cells_annot.shape)   
24. print(cancer_expr.shape, cancer_annot.shape) 
25. 
26. # adding missing cell types
27. **cell_types = CellTypes.load('configs/cell_types.yaml')**
28. missing_cts = [x for x in cell_types.get_all_subtypes('General_cells') if not x in cells_annot['Cell_type'].unique()]
29. 
30. for ct in missing_cts:
31.     subtypes = cell_types.get_direct_subtypes(ct)
32.     annot = cells_annot.loc[cells_annot['Cell_type'].isin(subtypes)]
33.     annot.index
34.     expr = cells_expr[annot.index]
35.     annot['Cell_type'] = ct
36.     annot.index = annot.index + f'_{ct}'
37.     annot['Dataset'] = annot.index
38.     expr.columns = expr.columns + f'_{ct}'
39.     cells_expr = pd.concat([cells_expr, expr], axis=1)
40.     cells_annot = pd.concat([cells_annot, annot])
41. 
42. # to make sure that there is no repeated samples
43. samples = sorted(list(set(cells_annot.index).intersection(set(cells_expr.columns))))
44. cells_expr = cells_expr[samples]
45. cells_annot = cells_annot.loc[samples]
46. print(cells_expr.shape, cells_annot.shape)
47. 
48. 
49. # Model training
50. mixer = Mixer(cell_types=cell_types,
51.               cells_expr=cells_expr, cells_annot=cells_annot,
52.               tumor_expr=cancer_expr, tumor_annot=cancer_annot,
53.               num_av=3, num_points=300000)
54. 
55. model = DeconvolutionModel(cell_types,
56.                            boosting_params_first_step='configs/boosting_params/lgb_parameters_first_step.tsv',
57.                            boosting_params_second_step='configs/boosting_params/lgb_parameters_second_step.tsv')
58. 
59. model.fit(mixer)

<img width="1306" height="250" alt="Image" src="https://github.com/user-attachments/assets/6c2cd5ce-9300-441e-b177-fa9bcef2eb68" />

Hello, I want to generate a tumor model. In the above code, I changed "Blood_model_annot" in line 1 to "Tumor_model_annot", made modifications in line 8, and changed line 27 to "cell_types.yaml". I didn't make any changes elsewhere. Is this acceptable? After execution, an error occurred at the step of model.fit(mixer), as shown in the figure. Could you please tell me the reason and how I should modify it?
